packages/vue/src/components/atoms/SfIcon/SfIcon.vue

```diff
@@ -82,70 +66,53 @@ export default {
       default: 1,
     },
   },
-  iconColorClass(color) {
-    const isSFColors = SF_COLORS.includes(color.trim());
-    return isSFColors ? `color-${color.trim()}` : "";
-  },
-  iconSizeClass(size) {
-    const isSFSizes = SF_SIZES.includes(size.trim());
-    if (isSFSizes) {
-      switch (size.trim()) {
-        case "xxs":
-          return "size-xxs";
-        case "xs":
-          return "size-xs";
-        case "sm":
-          return "size-sm";
-        case "md":
-          return "size-md";
-        case "lg":
-          return "size-lg";
-        case "xl":
-          return "size-xl";
-        case "xxl":
-          return "size-xxl";
-        case "xl3":
-          return "size-xl3";
-        case "xl4":
-          return "size-xl4";
-        default:
-          return "size-lg";
-      }
-    } else {
-      return "";
-    }
-  },
-  iconCustomStyle(color, size) {
-    const isSFColors = SF_COLORS.includes(color.trim());
-    const isSFSizes = SF_SIZES.includes(size.trim());
-    return {
-      "--icon-color": !isSFColors ? color : "",
-      "--icon-size": !isSFSizes ? size : "",
-    };
-  },
-  iconViewBox(icon, viewBox) {
-    const isSFIcons = () => {
-      if (typeof icon === "string") {
-        return SF_ICONS.includes(icon.trim());
-      } else return null;
-    };
-    return isSFIcons() ? icons[icon].viewBox || viewBox : viewBox;
-  },
-  iconPaths(icon) {
-    const isSFIcons = () => {
-      if (typeof icon === "string") {
-        return SF_ICONS.includes(icon.trim());
+  computed: {
+    isSFColors() {
+      return SF_COLORS.includes(this.color.trim());
+    },
+    isSFSizes() {
+      const size = this.size.trim();
+      return SF_SIZES.includes(size);
+    },
+    iconColorClass() {
+      return this.isSFColors ? `color-${this.color.trim()}` : "";
+    },
+    iconSizeClass() {
+      return this.isSFSizes ? `size-${this.size.trim()}` : "";
+    },
```

Comment:
Do you think static classes should be replaced with dynamic ones?

Reply:
@AdamPawlinski it's better for tests - with switch we should cover each case of switch. Also, there is a lot of repetitive code with those 'color-' and 'size-' prefixes that can be replaced with single line.